lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
> Things like 0x25 and 1E42 will convert to a narrower range of accepted numeric literal format, i.e. 37 and 1e+42 respectively

I didn't know that. 

Can you add test cases with a hex number and a floating point property with a capital `E`?

> I only based this upon my findings on REPL within Node.js - and I'm (overly) confident with my RegExp skills. 😎

+1